### PR TITLE
Port coverage report generation to pytest

### LINKED ISSUE #976: Switch Python Testing Package to Pytest
**What is the feature and why do you need it**:
Currently we are using pytest to test our clients and nose to generate coverage reports for the base repo. The reports can instead be generated with pytest as shown in [here](https://github.com/codecov/example-python).
**Describe the solution you'd like to see**:
Use pytest instead of nose to generate code coverage reports.
/good-first-issue
/help

### PR DESCRIPTION
**Pull request by palnabarun**
Fixes #976

### CODE REVIEW

**Comment by palnabarun** [comment]
cc @micw523

**Comment by micw523** [comment]
The coverage test is to provide a coverage report for codecov-io bot. It looks like you're running more tests than you need to - could you fix those? The tests in kubernetes/test does not have to be run for the coverage.

File: tox.ini [hunk 1]
```diff
@@ -22,27 +22,27 @@ commands =
...
 commands =
    python -V
-   {toxinidir}/scripts/kube-init.sh py.test -vvv -s []
+   {toxinidir}/scripts/kube-init.sh pytest -vvv -s []
 
 [testenv:coverage]
 commands =
    python -V
-   nosetests --with-coverage --cover-package=kubernetes.config,kubernetes.watch --cover-tests
+   pytest --ignore=kubernetes/e2e_test --cov=kubernetes/watch --cov=kubernetes/config
```
Comment (micw523) [inline tox.ini]: It looks like the tests for the dynamic client in base/dynamic are all e2e tests, which will be skipped by using pytest without a local cluster. If we are only testing the coverage for the watch and config folders we can skip those tests (for now).

**Comment by palnabarun** [comment]
Okay. I got it. I was assuming that `kubernetes/test` also has the tests for `k/config` and `k/watch`. Saw that they are inside the directories itself.

**Comment by palnabarun** [comment]
One small caveat that I bumped into though. When I limited the test run to `kubernetes/config` and `kubernetes/watch`, there was one line of code that was not covered by tests by those test.

image

The test that I ran:
```
pytest --cov=kubernetes/watch --cov=kubernetes/config kubernetes/watch kubernetes/config
```

@micw523 Is it acceptable to have that?

**Comment by palnabarun** [comment]
Updates: When I saw the coverage on the Travis CI interface, the very same line seems to be covered now. Maybe I have some inconsistencies in my dev environment.

**Comment by micw523** [comment]
/lgtm
Please squash your commits

**Comment by palnabarun** [comment]
@micw523 Done. Please approve. :)

**Comment by palnabarun** [comment]
Thank you @micw523 @yliaog :)

**Pull request merged**